swap/src/bitcoin/wallet.rs

```diff
@@ -195,7 +262,10 @@ impl WaitForTransactionFinality for Wallet {
     }
 }
 
+#[async_trait]
 impl GetNetwork for Wallet {
+    // todo: Get this from the electrum and instead of storing it in the Wallet
```

Comment:
Unless you really want to go all the way to electrum, the bdk-wallet itself has a function to get the network.  
```
self.inner.lock().await.network()
```